File: lesson_20/src/main/java/lesson20/service/CustomerService.java
package lesson20.service;

import lesson20.entities.Customer;
import lesson20.repository.CustomerRepository;
import org.springframework.stereotype.Component;

import java.util.List;

@Component
public class CustomerService {
    CustomerRepository customerRepository;
    CustomerService(CustomerRepository customerRepository) {
        this.customerRepository = customerRepository;
    }
    public void create(Customer customer) {
        customerRepository.create(customer);
    }

    public Customer getById(Long id) {
        return customerRepository.getById(id);
    }

    public List<Customer> getAll() {
        return customerRepository.getAll();
    }
}

File: lesson_20/src/main/kotlin/lesson20kt/services/CustomerService.kt
package lesson20kt.services

import lesson20kt.entities.Customer
import lesson20kt.repositories.CustomerRepository
import org.springframework.stereotype.Component

@Component
class CustomerService(val customerRepository: CustomerRepository) {

    fun save(customer: Customer) = customerRepository.save(customer)
    fun load(id: Long): Customer = customerRepository.load(id)
    fun loadAll(): List<Customer> = customerRepository.loadAll()
}